### ISSUE TITLE
Inability to GC CSI Plugins

### ISSUE THREAD

**Issue by eightseventhreethree** [hashicorp/nomad#11114]
### Nomad version
```
root@nomad-server-shared-001:/etc/nomad.d# nomad -version
Nomad v1.1.3 (8c0c8140997329136971e66e4c2337dfcf932692)
```

### Operating system and Environment details
```
root@nomad-server-shared-001:/etc/nomad.d# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

### Issue
Inability to GC old CSI plugins
```
root@nomad-server-shared-001:/etc/nomad.d# nomad plugin status -verbose
Container Storage Interface
ID                                    Provider         Controllers Healthy/Expected  Nodes Healthy/Expected
aws-ebs-${GD_AZ}                      ebs.csi.aws.com  0/0                           0/1
aws-ebs-${attr.platform.aws.placemen  ebs.csi.aws.com  0/0                           0/1
aws-ebs-${meta.GD_AZ}                 ebs.csi.aws.com  0/0                           0/1
aws-ebs-us-east-1b                    ebs.csi.aws.com  1/1                           1/1
aws-ebs-us-east-1c                    ebs.csi.aws.com  0/0                           1/1
aws-ebs0                              ebs.csi.aws.com  0/0                           0/1
```
```
root@nomad-server-shared-001:/etc/nomad.d# nomad system gc
root@nomad-server-shared-001:/etc/nomad.d# 
```
```
2021-09-01T16:13:55.514Z [DEBUG] worker: dequeued evaluation: eval_id=46075d9c-231c-00d3-53d3-deeab56064ab
2021-09-01T16:13:55.514Z [DEBUG] core.sched: forced job GC
2021-09-01T16:13:55.514Z [DEBUG] core.sched: forced eval GC
2021-09-01T16:13:55.514Z [DEBUG] core.sched: eval GC found eligibile objects: evals=1 allocs=0
2021-09-01T16:13:55.527Z [DEBUG] core.sched: forced deployment GC
2021-09-01T16:13:55.527Z [DEBUG] core.sched: forced plugin GC
2021-09-01T16:13:55.527Z [DEBUG] core.sched: CSI plugin GC scanning before cutoff index: index=18446744073709551615 csi_plugin_gc_threshold=1h0m0s
2021-09-01T16:13:55.551Z [DEBUG] core.sched: forced volume claim GC
2021-09-01T16:13:55.551Z [DEBUG] core.sched: CSI volume claim GC scanning before cutoff index: index=18446744073709551615 csi_volume_claim_gc_threshold=5m0s
2021-09-01T16:13:55.551Z [DEBUG] core.sched: forced node GC
2021-09-01T16:13:55.551Z [DEBUG] worker: ack evaluation: eval_id=46075d9c-231c-00d3-53d3-deeab56064ab
```

```
root@nomad-server-shared-001:/etc/nomad.d# nomad plugin status -verbose
Container Storage Interface
ID                                    Provider         Controllers Healthy/Expected  Nodes Healthy/Expected
aws-ebs-${GD_AZ}                      ebs.csi.aws.com  0/0                           0/1
aws-ebs-${attr.platform.aws.placemen  ebs.csi.aws.com  0/0                           0/1
aws-ebs-${meta.GD_AZ}                 ebs.csi.aws.com  0/0                           0/1
aws-ebs-us-east-1b                    ebs.csi.aws.com  1/1                           1/1
aws-ebs-us-east-1c                    ebs.csi.aws.com  0/0                           1/1
aws-ebs0                              ebs.csi.aws.com  0/0                           0/1
```
```
root@nomad-server-shared-001:/etc/nomad.d# nomad volume status
Container Storage Interface
No CSI volumes
```

### Reproduction steps
1. Register multiple CSI plugins
2. See plugins: `nomad plugin status`
2. Make sure all volumes are unregistered.
3. Run `nomad system gc`
4. Observe that plugins that have no references w/ either controllers or nodes are still registered.

#### Expected Result
Plugins w/ no references are pruned during manual GC run.

#### Actual Result
Plugins are staying around indefinitely.

**Comment by lgfa29** [comment]
Hi @eightseventhreethree 👋 

Are the plugin jobs still running? If so, could you try stopping then and running the GC.

Thanks!

**Comment by eightseventhreethree** [comment]
@lgfa29 The plugin jobs are stopped.

**Comment by tgross** [comment]
It looks like the plugins still have a reference, which is why `Expected: 0/1` for those plugins. But if the jobs are stopped, the count should be zero for sure. This doesn't look like a case of https://github.com/hashicorp/nomad/issues/8628, but I know we've had plugin count issues before that should have been resolved by 1.1.3, so we'll have to go back to the drawing board on that.

In the meantime, if you purge the stopped plugin jobs, do the plugins then get GC'd by any chance?

**Comment by tgross** [comment]
I'm going to close this as a duplicate of https://github.com/hashicorp/nomad/issues/11758 (it's slightly newer so technically that's the duplicate, but there's a bit more info and crosslinks there. Follow from https://github.com/hashicorp/nomad/issues/11758#issuecomment-1029211330 for more information.
